clang/lib/Driver/Driver.cpp

```diff
@@ -1242,7 +1242,13 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
           // the following iterations.
           FoundNormalizedTriples[NormalizedName] = Val;
           llvm::Triple DeviceTriple(MakeSYCLDeviceTriple(UserTargetName));
-          UniqueSYCLTriplesVec.push_back(DeviceTriple);
+          // Only add the Triple to the Unique vector if it is truly unique.
+          // We can have duplicate triples in cases where multiple implied
+          // targets are used, i.e. intel_gpu_pvc,intel_gpu_gen11
+          if (std::find(UniqueSYCLTriplesVec.begin(),
+                        UniqueSYCLTriplesVec.end(),
+                        DeviceTriple) == UniqueSYCLTriplesVec.end())
+            UniqueSYCLTriplesVec.push_back(DeviceTriple);
```

Comment:
Probably `UniqueSYCLTriplesVec` variables should be a set-like container instead of SmallVector 
- https://llvm.org/docs/ProgrammersManual.html#set-like-containers-std-set-smallset-setvector-etc.
StringSet sounds like a good candidate for that.

Reply:
Yep.